Document 2026's April Fools joke (#14612)
ppy/osu-wiki: wiki/History_of_osu!/April_Fools/en.md

April Fools' day 2021 recycled the same joke from the previous two years, which was a 1 in 20 chance of hearing a sound effect of a high-pitched girl sneezing upon opening a beatmap.[^sneeze-2-forums][^sneeze-2-forums-2]

## 2026

### More toggles

::: Infobox

> ![](img/2026-toggles.png "Toggles")

Reviewer: missing alt text for when the image can't be rendered

```suggestion
![Toggles section visible during April Fools' day](img/2026-toggles.png "Toggles section visible during April Fools' day")
```

:::

> In the options sidebar, there was a new `Toggles` section accessible in osu!(lazer) on 1 April 2026[^2026], featuring a text saying "You wanted toggles, we give you toggles" and a single toggle below. This toggle would cause new toggles to appear when clicked. Silly labels on the toggles like "toggle chaos" or "toggle entropy" added to the comedic effect.

Reviewer: footnote formatting is special (https://osu.ppy.sh/wiki/en/Article_styling_criteria/Formatting#footnotes)

```suggestion
In the options sidebar, there was a new `Toggles` section accessible in [osu!(lazer)](/wiki/Client/Release_stream/Lazer) on 1 April 2026,[^2026] featuring text saying "You wanted toggles, we give you toggles" and a single toggle below. This toggle caused new toggles to appear when clicked, labelled with silly names like "toggle chaos" or "toggle entropy" for comedic effect.
```

> Activating any toggle would often select or deselect other toggles and apply random effects on the game client, some of which would stay effective while playing a beatmap. Some of the effects the users were able to encounter when clicking on a toggle:

Reviewer: ```suggestion
Activating any toggle would select or deselect other toggles and apply random effects, some of which stayed effective even while playing a beatmap:
```

- an osu! icon moving around the screen like the [DVD logo screensaver](https://en.wikipedia.org/wiki/DVD_screensaver)
- osu! doing a barrel roll[^barrel-roll]
- the menu interface having a new colour design
- various sound effects such as distorted audio

> - the game window being slightly tilted

Reviewer: ```suggestion
- An osu! icon moving around the screen like the [DVD logo screensaver](https://en.wikipedia.org/wiki/DVD_screensaver)
- osu! doing a barrel roll[^barrel-roll]
- The menu interface having a new colour scheme
- Various sound effects such as distorted audio
- The game window being slightly tilted
```

> When clicking too many toggles, osu! eventually would play a crash animation by zooming in and fading to black. Following that, the game was accessible again, however, in the `Toggles` section there was now a text saying "..on second thought, no more toggles for you." A button with the text "But please peppy" below that would play a buzzer sound and display the text "no" when clicked. After pressing the button enough times, the dialogue would ultimately reset itself, displaying one toggle as described above again.

Reviewer: slightly odd word choice:
- game was always "accessible" (you could play beatmaps for example)
- there was no popup "dialogue" anywhere

```suggestion
When clicking too many toggles, osu! would eventually "crash" by playing a sound of glass shattering, zooming in and fading to black, then return to a normal state. However, the `Toggles` section would then have text saying "..on second thought, no more toggles for you.", and a button below that labelled "But please peppy" would play a buzzer sound and display the text "no" when clicked. Pressing the button enough times would reset the `Toggles` section and display one random toggle as described above.
```

> This new section in the options menu representing placeholders for actual settings serves as a reference to the many features desired by users as seen on [GitHub discussions](https://github.com/ppy/osu/discussions) and in the [Feature Requests](https://osu.ppy.sh/community/forums/4) forum. The [developers](/wiki/People/Developers) want to prevent the options menu from becoming too overloaded by moving the settings elsewhere in the user interface.

Reviewer: the last part isn't a general case - often new toggles would cause more clutter than being useful to the vast majority of the playerbase, for example

also a citation to a statement/response by any of the developers would be nice here

```suggestion
This toggle section was added in reference to the many requests on [GitHub discussions](https://github.com/ppy/osu/discussions) and in the [Feature Requests](https://osu.ppy.sh/community/forums/4) forum that have proposed adding toggles for various features. The [developers](/wiki/People/Developers) want to prevent the options menu from becoming too overloaded.
```

[^sneeze-2-reddit-2]: [Reddit post by u/ohmaytt in r/osugame (2020-04-01) "This year's osu! April Fool's Day joke"](https://www.reddit.com/r/osugame/comments/fsq30l/this_years_osu_april_fools_day_joke/)
[^sneeze-2-forums]: [Forum thread by MilkyIQ (2021-04-01) "Is this not the third year in a row that we get sneezing girl?"](https://osu.ppy.sh/community/forums/topics/1286906)
[^sneeze-2-forums-2]: [Forum thread by GreatTurtleKing (2021-04-01) "i heard like a sneeze when i just started to play a song"](https://osu.ppy.sh/community/forums/topics/1286396)

> [^2026]: [osu!(lazer) source code on GitHub](https://github.com/ppy/osu/commit/3edc428c3a084896ec6fcb4f09528cdd0668c0ef)

Reviewer: https://github.com/ppy/osu-wiki/blob/master/wiki/Beatmap_search/en.md?plain=1#L160-L161

```suggestion
[^2026]: [osu!(lazer) source code](https://github.com/ppy/osu/commit/3edc428c3a084896ec6fcb4f09528cdd0668c0ef)
```

> [^barrel-roll]: A reference to the famous Google easter egg, where the website would roll over when typing "do a barrel roll" into the search bar.

Reviewer: where did you get this from? the source code comment clearly indicates that it's a star fox reference

https://github.com/ppy/osu/commit/3edc428c3a084896ec6fcb4f09528cdd0668c0ef#diff-8b5b87382dcb87ebe3b577168cfa71ce81baaf6d5d0762990954785ad34e44f5R317

>  // do a barrel roll! (press Z or R twice)

<img width="620" height="349" alt="Image" src="https://github.com/user-attachments/assets/45c6b0a6-0d5a-4bbc-9f7e-75f23fa1e515" />

```suggestion
[^barrel-roll]: A reference to a meme from [Star Fox 64](https://en.wikipedia.org/wiki/Star_Fox_64), where Peppy Hare advises the player to "Do a barrel roll!", i.e. rotate 360 degrees.
```

[^sneeze-2-reddit-2]: [Reddit post by u/ohmaytt in r/osugame (2020-04-01) "This year's osu! April Fool's Day joke"](https://www.reddit.com/r/osugame/comments/fsq30l/this_years_osu_april_fools_day_joke/)
[^sneeze-2-forums]: [Forum thread by MilkyIQ (2021-04-01) "Is this not the third year in a row that we get sneezing girl?"](https://osu.ppy.sh/community/forums/topics/1286906)
[^sneeze-2-forums-2]: [Forum thread by GreatTurtleKing (2021-04-01) "i heard like a sneeze when i just started to play a song"](https://osu.ppy.sh/community/forums/topics/1286396)

[^2026]: [osu!(lazer) source code](https://github.com/ppy/osu/commit/3edc428c3a084896ec6fcb4f09528cdd0668c0ef)
[^barrel-roll]: A reference to a meme from [Star Fox 64](https://en.wikipedia.org/wiki/Star_Fox_64), where Peppy Hare advises the player to "Do a barrel roll!", i.e. rotate 360 degrees.

> [^toggles-comment]: [GitHub comment by peppy (2023-04-16)](https://github.com/ppy/osu/pull/23200#issuecomment-1510138723)

Author: For your information, next to the PR linked in this reference, I was able to find other PRs trying to implement the very same feature:

- https://github.com/ppy/osu/pull/24947
- https://github.com/ppy/osu/pull/27285
